app/controllers/signup.js

```diff
@@ -1,194 +1,218 @@
     {
-      id: 'github',
-      label: 'Github ',
-      type: 'text',
-      errorMessage: 'GitHub username is required',
+      id: "github_id",
+      label: "Github_id ",
+      type: "text",
+      errorMessage: "GitHub username is required",
       required: true,
-      showError: false
+      showError: false,
     },
     {
-      id: 'linkedIn',
-      label: 'LinkedIn ',
-      type: 'text',
-      errorMessage: 'LinkedIn username is required',
+      id: "linkedin_id",
+      label: "linkedin_id ",
+      type: "text",
+      errorMessage: "linkedIn username is required",
       required: true,
-      showError: false
+      showError: false,
     },
     {
-      id: 'twitter',
-      label: 'Twitter ',
-      type: 'text',
-      errorMessage: 'Twitter handle is required',
+      id: "twitter_id",
+      label: "twitter_id ",
+      type: "text",
+      errorMessage: "Twitter handle is required",
       required: true,
-      showError: false
+      showError: false,
     },
     {
-      id: 'website',
-      label: 'Website ',
-      type: 'text',
-      errorMessage: '',
+      id: "website",
+      label: "Website ",
+      type: "text",
+      errorMessage: "",
       required: false,
-      showError: false
+      showError: false,
     },
-  ]
+  ];
 
   @action handleFieldChange(name, value) {
-    const index = this.fields.findIndex(field => field.id === name)
-    
-    set(this.formData, name, value)
-    
+    const index = this.fields.findIndex((field) => field.id === name);
+    set(this.formData, name, value);
+
     if (this.fields[index].required && !value) {
-      this.isSubmitDisabled = true
-      return set(this.fields[index], 'showError', true)
+      this.isSubmitDisabled = true;
+      set(this.fields[index], "showError", true);
     } else if (!this.fields[index].validator) {
-      return set(this.fields[index], 'showError', false)
+      set(this.fields[index], "showError", false);
     }
 
-    const anyErrors = this.fields.reduce((prev, field) => {
-      if (prev) {
-        return prev
+    const anyErrors = this.fields.map((field) => {
+      if (field.required && this.formData[field.id] === "") {
+        return true;
       }
-      if (field.required && this.formData[field.id] === '') {
-        return true
-      }
-      return false
-    }, false)
+      return false;
+    });
 
-    if (anyErrors) {
-      this.isSubmitDisabled = true
+    if (anyErrors.filter(Boolean).length) {
+      this.isSubmitDisabled = true;
+    } else {
+      this.isSubmitDisabled = false;
     }
   }
 
-  @action phoneNumberValidator(phoneNumber) {
-    if (typeof phoneNumber !== 'string'){ return false }
+  @action phoneNumberValidator(phone_number) {
+    if (typeof phone_number !== "string") {
+      return false;
+    }
 
-    const pattern=/^(0|[+91]{3})?[7-9][0-9]{9}$/;
-    const index = this.fields.findIndex((field) => field.id === 'phoneNumber')
+    const pattern = /^(0|[+91]{3})?[7-9][0-9]{9}$/;
```

Comment:
This will accept only IN numbers. Wouldn't it be nice to provide support for international numbers also?😅

Reply:
I guess then in that case, there wouldn't be any pattern. We need to put a country field too, based on that user's country we need to apply the pattern.  Also, the best case would be here to use any third party library.
I guess for right now we can leave it as it is, in the future we can refactor based on use cases.